src/Lib.P2P/Discovery/MdnsJs.cs

```diff
@@ -0,0 +1,108 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Makaretu.Dns;
+using MultiFormats;
+
+namespace Lib.P2P.Discovery
+{
+    /// <summary>
+    ///   Discovers peers using Multicast DNS according to
+    ///   js-ipfs v0.32.3
+    /// </summary>
+    public class MdnsJs : Mdns
+    {
+        /// <summary>
+        ///   Creates a new instance of the class.  Sets the <see cref="Mdns.ServiceName"/>
+        ///   to "ipfs".
+        /// </summary>
+        public MdnsJs() { ServiceName = "ipfs"; }
+
+        /// <inheritdoc />
+        protected override void OnServiceDiscovery(ServiceDiscovery discovery)
+        {
+            discovery.AnswersContainsAdditionalRecords = true;
+        }
+
+        /// <inheritdoc />
+        public override ServiceProfile BuildProfile()
+        {
+            // Only internet addresses.
+            var addresses = LocalPeer.Addresses
+               .Where(a => a.Protocols.First().Name == "ip4" || a.Protocols.First().Name == "ip6")
+               .ToArray();
+            if (addresses.Length == 0) return null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)